### PR TITLE
Add Medium icon

### PR DESCRIPTION
**Pull request by jasmussen**
As requested by a user.

### CODE REVIEW

File: svg-min/medium.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><rect x="0" fill="none" width="24" height="24"/><g><style>.st0{fill:#231F20;}</style><path class="st0" d="M20.9 7.2l-5.4-2.7h-.1-.1c-.2 0-.4.1-.5.3l-3.1 5.1 3.9 6.4 5.3-9.1c.1 0 .1 0 0 0zM9.9 8.6v5.8l5.1 2.5zM15.8 17.3l4.2 2.1c.6.3 1 .1 1-.4V8.8l-5.2 8.5zM9 7.2L3.7 4.5c-.1 0-.2-.1-.3-.1-.2.1-.4.2-.4.5v11.4c0 .3.2.7.5.8l4.7 2.3c.1.1.2.1.3.1.3.1.5-.1.5-.5V7.2z"/></g></svg>
```
Comment (aduth) [inline svg-min/medium.svg]: We should remove this `id="Layer_1"` and anywhere else it affects (generated files included).

**Comment by jasmussen** [comment]
Thanks a bunch @aduth for that catch, embarrassing that that snuck in. I've updated the Gruntfile to cleanup IDs, as you say I don't think we'll need them in any of the files.

File: svg-min/medium.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><rect x="0" fill="none" width="24" height="24"/><g><g><path d="M20.9 7.2l-5.4-2.7h-.2c-.2 0-.4.1-.5.3l-3.1 5.1 3.9 6.4 5.3-9.1c.1 0 .1 0 0 0zM9.9 8.6v5.8l5.1 2.5M15.8 17.3l4.2 2.1c.6.3 1 .1 1-.4V8.8l-5.2 8.5zM9 7.2L3.7 4.5c-.1 0-.2-.1-.3-.1-.2.1-.4.2-.4.5v11.4c0 .3.2.7.5.8l4.7 2.3c.1.1.2.1.3.1.3.1.5-.1.5-.5V7.2z"/></g></g></svg>
```
Comment (aduth) [inline svg-min/medium.svg]: Silly small, but nested `<g>` is redundant :smile: Not to hold up a merge, but minor clean-up.

Comment (aduth) [inline svg-min/medium.svg]: Oh, I guess it's the optimizer adding this? Strange...

**Comment by aduth** [comment]
Can't speak to other use-cases, but dropping the React components in Calypso tests well for me. :+1: on that front.

File: svg-min/medium.svg (hunk 2 above)
Comment (jasmussen) [inline svg-min/medium.svg]: > Oh, I guess it's the optimizer adding this? Strange...

This was intentionally added to Gridicons and therefore inherited to Social Logos when we forked. The purpose of the extra `g` tag is to be able to specify it for CSS (i.e. `.classname svg g`). This is necessary in the case of subpixel `transform` operations without interfering with any `transform`s you might want to add as part of animations or something.

Comment (aduth) [inline svg-min/medium.svg]: >The purpose of the extra `g` tag is to be able to specify it for CSS

But is it necessary to have two? Right now it's `<g><g> ... </g></g>`

Comment (jasmussen) [inline svg-min/medium.svg]: Doh! No, that's definitely a bug. Great catch, I'll look about fixing that!

**Comment by jasmussen** [comment]
Fixed issue where the Amazon and Medium logos had redundant `g` elements in them. I'm not entirely sure exactly how they snuck in — I suspect Illustrator baked some weird groups in the SVG source files, and we ran into limitations of svg-min. But nonetheless this should be fixed.

**Comment by aduth** [comment]
Looking good in Calypso :+1:

**Pull request merged**